cirq-google/cirq_google/engine/engine_program.py

```diff
@@ -575,11 +402,11 @@ async def batch_size_async(self) -> int:
                 self.project_id, self.program_id, True
             )
         code = self._program.code
-        code_type = code.type_url[len(engine_base.TYPE_PREFIX) :]
-        if code_type == 'cirq.google.api.v2.BatchProgram':
-            batch = v2.batch_pb2.BatchProgram.FromString(code.value)
-            return len(batch.programs)
-        raise ValueError(f'Program was not a batch program but instead was of type {code_type}.')
+        code_type = code.type_url[len(engine_base.TYPE_PREFIX) :]  # pragma: no cover
+        raise ValueError(  # pragma: no cover
```

Comment:
Can we remove `batch_size_async` altogether? If not, can we remove its implementation and raise a [`NotImplementedError`](https://docs.python.org/3/library/exceptions.html#NotImplementedError)?